tests/trace_server/test_calls_query_builder.py

```diff
@@ -1100,21 +1100,18 @@ def test_calls_query_with_complex_heavy_filters() -> None:
         FROM calls_merged
         WHERE
             calls_merged.project_id = {pb_1:String}
-        AND
-            (calls_merged.id IN filtered_calls)
-        AND (((calls_merged.inputs_dump LIKE {pb_10:String} OR calls_merged.inputs_dump IS NULL))
-        AND (((JSON_VALUE(calls_merged.inputs_dump, {pb_4:String}) > {pb_5:UInt64}) OR calls_merged.inputs_dump IS NULL))
-        AND ((calls_merged.output_dump LIKE {pb_11:String} OR calls_merged.output_dump IS NULL))
-        AND ((lower(calls_merged.output_dump) LIKE {pb_12:String} OR calls_merged.output_dump IS NULL)))
+          AND (calls_merged.id IN filtered_calls)
+          AND (
```

Comment:
reduction of conditions here is intentional?

Reply:
The reduction is technically a regression, we are no longer doing pre-having optimizations for integer comparisons. I'll make a ticket to add it back. The current impl focuses on strings, which dominates the current bad-query numbers.